ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilder.java

```diff
@@ -221,6 +233,28 @@ private static PhysicalSchema getPhysicalSchemaWithKeyAndPseudoCols(
     );
   }
 
+  private static PhysicalSchema getPhysicalSchemaWithPseudoColumnsToMaterialize(
+      final SourceStep<?> streamSource) {
+
+    final FormatInfo f = streamSource.getFormats().getKeyFormat();
+    final SerdeFeatures s = streamSource.getFormats().getKeyFeatures();
+    final KeyFormat k = streamSource instanceof WindowedTableSource
+        ? KeyFormat.windowed(f, s, ((WindowedTableSource) streamSource).getWindowInfo())
+        : KeyFormat.nonWindowed(f, s);
```

Comment:
This should, in it's current state, never be called on WindowedTableSources. While it's probably a bug that it isn't, considering joins on windowed table sources don't even work, I think I should remove this for now to reduce code complexity. WDYT @vcrfxia?

Reply:
If this block of code is not meant to be called on windowed table sources, then I'd have the top of the method throw an exception if a windowed table source is encountered and then simplify the body of the code with the assumption that the method is not called with windowed table sources.

Reply:
The method takes a `SourceStep`, so it would be equally wrong to give it a `StreamSource` or `WindowedStreamSource`. The only reason to explicitly disallow `WindowedTableSource` would be because we don't work with the stream steps in this file- I think throwing an exception on a `WindowedTableSource` might confuse someone down the line, as I think the method could eventually be used with it, and explicitly adding an exception might give someone the idea that there is a greater reason for not accepting `WindowedTableSource`s. Because of this, maybe I could just make the method accept a `TableSource` and cast `source` where it's called to a `TableSource{V2}`. Do you agree with my train of thought?